plugins/module_utils/foreman_helper.py

```diff
@@ -178,6 +181,28 @@ def __init__(self, **kwargs):
         super(TaxonomyMixin, self).__init__(foreman_spec=foreman_spec, **kwargs)
 
 
+class ParametersMixin(object):
+    def __init__(self, **kwargs):
+        self.entity_name = kwargs.pop('entity_name', self.entity_name_from_class)
+        foreman_spec = dict(
+            parameters=dict(type='list', elements='dict', options=parameter_ansible_spec, flat_name='{0}_parameters_attributes'.format(self.entity_name)),
```

Comment:
`'{0}_parameters_attributes'` This naming convention is like 0.6 accurate?

how about
```python
parameters_flat_name = getattr(self, "PARAMETERS_FLAT_NAME") or '{0}_parameters_attributes'.format(self.entity_name)
```
And you can add that as a class constant when needed.